components/icon-button.jsx

```diff
@@ -4,18 +4,23 @@ var Link = Router.Link;
 var ImageTag = require('./imagetag.jsx');
 
 var LinkSwap = React.createClass({
+  propTypes: {
+    linkTo: React.PropTypes.string,
+    href: React.PropTypes.string,
+    onClick: React.PropTypes.func
```

Comment:
So your first link is to this code:

```js
    var props = Object.assign({}, this.props, {
      href: router.makeHref(to, query),
      onClick: this.handleClick
    });
```

The `this.props` in the first line basically means it's copying all the properties passed to it, but then overriding the `href` and `onClick` properties with its own values.